index.js

```diff
@@ -111,6 +116,51 @@ Babel.prototype.optionsHash = function() {
       hash[key] = (typeof value === 'function') ? (value + '') : value;
     }
 
+    if (options.plugins) {
+      hash.plugins = [];
+
+      for (var i = 0; i < options.plugins.length; i++) {
+        var plugin = options.plugins[i];
+
+        var pluginType = typeof plugin;
+        var pluginAugmentsCacheKey = false;
+        var pluginProvidesBaseDir = false;
+        var pluginRequiresBaseDir = true;
+
+        // do not look for properties on string plugins
+        if (Array.isArray(plugin)) {
+          // when an array, the second argument is an options hash
+          pluginRequiresBaseDir = false;
+          hash.plugins.push(stringify(plugin[1]));
```

Comment:
so in the array case, each member of the array is a 100% serializable "pure" object?